### PR TITLE
[PREVIEW] RDM-2774 Read documents from Azure Blob Store

### PR DESCRIPTION
**Pull request by rafalkalita**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-2774

### Change description ###
Read documents from Azure Blob Store if contentUri is present in db

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: application/src/main/java/uk/gov/hmcts/dm/controller/DocumentContentVersionController.java [hunk 1]
```diff
@@ -27,10 +28,14 @@
...
 import uk.gov.hmcts.dm.service.StoredDocumentService;
 
+import javax.servlet.http.HttpServletResponse;
 import javax.validation.Valid;
+import java.io.IOException;
 import java.util.UUID;
 import java.util.stream.Collectors;
 
+import static org.apache.commons.lang.StringUtils.isBlank;
+
```
Comment (adr1ancho) [inline application/src/main/java/uk/gov/hmcts/dm/controller/DocumentContentVersionController.java]: Shall we use the later `org.apache.commons.lang3.StringUtils`

File: application/src/main/java/uk/gov/hmcts/dm/controller/StoredDocumentController.java [hunk 2]
```diff
@@ -26,16 +28,20 @@
...
 import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletResponse;
 import javax.validation.Valid;
+import java.io.IOException;
 import java.util.List;
 import java.util.UUID;
 
+import static java.lang.String.format;
+import static org.apache.commons.lang.StringUtils.isBlank;
+
```
Comment (adr1ancho) [inline application/src/main/java/uk/gov/hmcts/dm/controller/StoredDocumentController.java]: Shall we use the later org.apache.commons.lang3

File: application/src/main/java/uk/gov/hmcts/dm/controller/StoredDocumentController.java [hunk 3]
```diff
@@ -99,30 +105,47 @@ void init() throws NoSuchMethodException {
...
     })
-    public ResponseEntity<Object> getBinary(@PathVariable UUID documentId) {
+    public ResponseEntity<Object> getBinary(@PathVariable UUID documentId, HttpServletResponse response) {
 
         DocumentContentVersion documentContentVersion =
-                documentContentVersionService.findMostRecentDocumentContentVersionByStoredDocumentId(documentId);
+            documentContentVersionService.findMostRecentDocumentContentVersionByStoredDocumentId(documentId);
 
         if (documentContentVersion == null || documentContentVersion.getStoredDocument().isDeleted()) {
             return ResponseEntity.notFound().build();
```
Comment (adr1ancho) [inline application/src/main/java/uk/gov/hmcts/dm/controller/StoredDocumentController.java]: Would you have a chance in this story to find out how documentContentVersion.getStoredDocument().isDeleted() is used and check whether we can reduce complexity to one single place

File: application/src/main/java/uk/gov/hmcts/dm/hateos/DocumentContentVersionHalResource.java [hunk 4]
```diff
@@ -45,7 +45,7 @@ public DocumentContentVersionHalResource(DocumentContentVersion documentContentV
 
         add(linkTo(methodOn(DocumentContentVersionController.class).getDocumentContentVersionDocumentBinary(
                 documentContentVersion.getStoredDocument().getId(),
-                documentContentVersion.getId())).withRel("binary"));
+                documentContentVersion.getId(),null)).withRel("binary"));
```
Comment (adr1ancho) [inline application/src/main/java/uk/gov/hmcts/dm/hateos/DocumentContentVersionHalResource.java]: I think you may need a space after comma or some tools may complain

File: application/src/main/java/uk/gov/hmcts/dm/service/DocumentContentVersionService.java [hunk 5]
```diff
@@ -30,30 +27,19 @@
...
     private StoredDocumentRepository storedDocumentRepository;
 
-    @Autowired
-    @Setter
-    private HttpServletResponse response;
-
-    @Setter
-    @Getter
-    private int streamBufferSize = 5000000;
+    private static int STREAM_BUFFER_SIZE = 5000000;
```
Comment (adr1ancho) [inline application/src/main/java/uk/gov/hmcts/dm/service/DocumentContentVersionService.java]: you mean private static final ?  or do you want to make it configurable later?

File: application/src/test/java/uk/gov/hmcts/dm/componenttests/ComponentTestConfiguration.java [hunk 6]
```diff
@@ -15,9 +13,9 @@
...
     }
 
-    @Bean
-    public SubjectResolver<User> userResolver() {
-        return new UserResolverBackdoor();
-    }
+//    @Bean
+//    public SubjectResolver<User> userResolver() {
+//        return new UserResolverBackdoor();
+//    }
```
Comment (adr1ancho) [inline application/src/test/java/uk/gov/hmcts/dm/componenttests/ComponentTestConfiguration.java]: Shall we remove these lines?

File: application/src/main/java/uk/gov/hmcts/dm/controller/DocumentContentVersionController.java (hunk 1 above)
Comment (rafalkalita) [reply application/src/main/java/uk/gov/hmcts/dm/controller/DocumentContentVersionController.java]: Will change

File: application/src/main/java/uk/gov/hmcts/dm/controller/StoredDocumentController.java (hunk 2 above)
Comment (rafalkalita) [reply application/src/main/java/uk/gov/hmcts/dm/controller/StoredDocumentController.java]: Will change

File: application/src/main/java/uk/gov/hmcts/dm/controller/StoredDocumentController.java (hunk 3 above)
Comment (rafalkalita) [reply application/src/main/java/uk/gov/hmcts/dm/controller/StoredDocumentController.java]: As per chat we have had we will leave it as it is for now.

File: application/src/main/java/uk/gov/hmcts/dm/service/DocumentContentVersionService.java (hunk 5 above)
Comment (rafalkalita) [reply application/src/main/java/uk/gov/hmcts/dm/service/DocumentContentVersionService.java]: yes and not configurable.

File: application/src/main/java/uk/gov/hmcts/dm/hateos/DocumentContentVersionHalResource.java (hunk 4 above)
Comment (rafalkalita) [reply application/src/main/java/uk/gov/hmcts/dm/hateos/DocumentContentVersionHalResource.java]: thanks

File: application/src/test/java/uk/gov/hmcts/dm/componenttests/ComponentTestConfiguration.java (hunk 6 above)
Comment (rafalkalita) [reply application/src/test/java/uk/gov/hmcts/dm/componenttests/ComponentTestConfiguration.java]: Yep

**Pull request merged**